mlflow/models/__init__.py

```diff
@@ -21,17 +21,26 @@
 For details, see `MLflow Models <../models.html>`_.
 """
 
-from .model import Model
-from .flavor_backend import FlavorBackend
-from .signature import ModelSignature, infer_signature
-from .utils import ModelInputExample
-from ..utils.environment import infer_pip_requirements
+from .evaluation import evaluate
 
 __all__ = [
-    "Model",
-    "ModelSignature",
-    "ModelInputExample",
-    "infer_signature",
-    "FlavorBackend",
-    "infer_pip_requirements",
+    "evaluate",
 ]
+
+try:
+    from .model import Model
+    from .flavor_backend import FlavorBackend
+    from .signature import ModelSignature, infer_signature
+    from .utils import ModelInputExample
+    from ..utils.environment import infer_pip_requirements
+
+    __all__ += [
+        "Model",
+        "ModelSignature",
+        "ModelInputExample",
+        "infer_signature",
+        "FlavorBackend",
+        "infer_pip_requirements",
+    ]
+except ImportError:
+    pass
```

Comment:
Why do we need this change?

Reply:
https://github.com/mlflow/mlflow/pull/5069#issuecomment-982880567

Reply:
@WeichenXu123 Can we include `infer_pip_requirements`, `Model`, and `FlavorBackend` unconditionally here - https://github.com/mlflow/mlflow/pull/5069/files#diff-433650164140d873b2734597cd9015aeb9f8bad5563230eaca543df1a117b8d4R27 and only try / catch `ModelInputSignature`, `infer_signature`, and `ModelInputExample`?

Reply:
@WeichenXu123 Can we add a test case ensuring that `mlflow.evaluate` is importable from the skinny client? For reference, our skinny test suites are run here: https://github.com/mlflow/mlflow/blob/master/dev/run-python-skinny-tests.sh